reframe/core/buildsystems.py

```diff
@@ -658,43 +659,45 @@ def emit_build_commands(self, environ):
 class EasyBuild(BuildSystem):
     easyconfigs = fields.TypedField(typ.List[str])
     options = fields.TypedField(typ.List[str])
-    installpath = fields.TypedField(str, type(None))
-    buildpath = fields.TypedField(str, type(None))
-    sourcepath = fields.TypedField(str, type(None))
-    use_system_config = fields.TypedField(bool)
+    emit_package = fields.TypedField(bool)
+    package_opts = fields.TypedField(typ.Dict[str, str])
 
     def __init__(self):
         super().__init__()
         self.easyconfigs = []
         self.options = []
-        self.installpath = None
-        self.buildpath = None
-        self.sourcepath = None
-        self.use_system_config = False
+        self.emit_package = False
+        self.eb_modules = []
 
     def emit_build_commands(self, environ):
         easyconfigs = ' '.join(self.easyconfigs)
+
+        if self.emit_package:
+            self.options.append('--package')
+            for key, val in self.package_opts.items():
+                self.options.append(f"--package-{key}={val}")
+
         options = ' '.join(self.options)
-        if self.use_system_config:
-            return [f"eb {easyconfigs} {options}"]
-
-        return [f"eb {easyconfigs} --installpath {self.installpath} "
-                # f"--tmpdir {self._installpath} "
-                f"--sourcepath {self.sourcepath} "
-                f"--buildpath {self.buildpath} {options}"]
-
-    def eb_modules(self):
-        modules = []
-        modules_dir = os.path.join(self.installpath, 'modules', 'all')
-        if os.path.isdir(modules_dir):
-            for mod in os.listdir(modules_dir):
-                for ver in os.listdir(os.path.join(modules_dir, mod)):
-                    modules.append({'name': f'{mod}/{ver}',
+
+        cmd = [f"EASYBUILD_BUILDPATH={self._eb_sandbox}/build \\",
+               f"EASYBUILD_INSTALLPATH={self._eb_sandbox} \\",
+               f"EASYBUILD_PREFIX={self._eb_sandbox} \\",
+               f"EASYBUILD_SOURCEPATH={self._eb_sandbox} \\",
+               f"eb {easyconfigs} {options}"]
+
+        return cmd
+
+    def collect_built_modules(self, build_stdout):
```

Comment:
The users should call this functions, but better rename it `generated_modules()`. You don't need the `eb_modules` attributes. In order to avoid opening and reading files every time, you can simply cache the result internally in the build system.